lib/shark_client.js

```diff
@@ -129,6 +128,26 @@ function _request(opts, cb) {
     }
 }
 
+/*
+ * Use information about the object to craft the storage path.
+ */
```

Comment:
Some comment or documentation on the storage path components should be *somewhere*. Could be on this function.

- why the hash
- doc that the hash is an md5
- why "2" for the objectIdPrefix

Having that info be a comment on this function would be reasonable.